src/MonoMod.Core/Platforms/Systems/MacOSArm64Helper.cs

```diff
@@ -0,0 +1,140 @@
+        {
+            _handle = DynDll.OpenLibrary(fileName);
+            
+            try
+            {
+                _jitMemCpy = DynDll.GetExport(_handle, JitMemCpyExport);
+                Helpers.Assert(_jitMemCpy != IntPtr.Zero);
+
+                _jitHookConfig = DynDll.GetExport(_handle, JitHookConfigExport);
+                Helpers.Assert(_jitHookConfig != IntPtr.Zero);
+            }
+            catch
+            {
+                DynDll.CloseLibrary(_handle);
+                throw;
+            }
+        }
+
+        /// <summary>
+        /// JIT_MAP (RWX) safe memory copy
+        /// </summary>
+        /// <param name="dst">Destination</param>
+        /// <param name="src">Source</param>
+        /// <param name="size">Size of memory that should be copied from dst to src</param>
+        public unsafe void JitMemCpy(IntPtr dst, IntPtr src, ulong size)
+        {
+            var fnPtr = (delegate* unmanaged[Cdecl]<IntPtr, IntPtr, ulong, void>)_jitMemCpy;
+            
+            fnPtr(dst, src, size);
+        }
+
+        /// <summary>
+        /// Gets the pointer to the native jit hook configuration struct which can vary by both runtime and arch.
+        /// </summary>
+        /// <param name="runtimeMajMin">Runtime major and minor version.</param>
+        /// <returns>A pointer to the requested jit hook configuration struct.</returns>
+        internal unsafe IntPtr GetJitHookConfig(int runtimeMajMin)
+        {
+            var fnPtr = (delegate* unmanaged[Cdecl]<int, IntPtr>)_jitHookConfig;
+            return fnPtr(runtimeMajMin);
+        }
+
+        private static void CreateTempFile(string tmpl, out int fd, out string fileName)
+        {
+            var tmplName = ArrayPool<byte>.Shared.Rent(tmpl.Length + 1);
+            
+            try
+            {
+                unsafe
+                {
+                    tmplName.AsSpan().Clear();
+                    Encoding.UTF8.GetBytes(tmpl).AsSpan().CopyTo(tmplName);
+
+                    fixed (byte* tmplNamePtr = tmplName)
+                        fd = OSX.MkSTemp(tmplNamePtr);
+
+                    if (fd == -1)
+                    {
+                        var lastError = OSX.Errno;
+                        var ex = new Win32Exception(lastError);
+                        MMDbgLog.Error($"Could not create temp file for {nameof(MacOSArm64Helper)}: {lastError} {ex}");
+                        throw ex;
+                    }
+
+                    fileName = Encoding.UTF8.GetString(tmplName, 0, tmpl.Length);
+                }
+            }
+            finally
+            {
+                ArrayPool<byte>.Shared.Return(tmplName);
+            }
+        }
+
+        private static void ExtractEmbeddedResourcesToTempFile(int fd, string logicalName)
+        {
+            using (var handle = new SafeFileHandle((IntPtr)fd, true))
+            using (var stream = new FileStream(handle, FileAccess.Write))

```

Comment:
`SafeFileHandle` will not work properly with P/Invoked file descriptor on Mono runtime. 
See: mono/mono/issues/12783